DietPi-Software | Moonlight: Game streaming client for Sunshine and NVIDIA GameStream

mtekman opened the pull request:
Add [Moonlight](https://github.com/moonlight-stream/moonlight-embedded/) game streaming client

This is the embedded version suitable for RPI, based on their debian bullseye build.

The installation and deinstallation script is based on the [setup script](https://dl.cloudsmith.io/public/moonlight-game-streaming/moonlight-embedded/setup.deb.sh) shown on their [packages page](https://github.com/moonlight-stream/moonlight-embedded/wiki/Packages).

In terms of documentation, I'm not sure what's really needed. People can just run `moonlight list` and it will automatically look for any broadcasting services and try to match and pair, as given in their [Usage section](https://github.com/moonlight-stream/moonlight-embedded/wiki/Usage) on their wiki.

mtekman committed:
added moonlight

mtekman committed:
tabify whitespace

mtekman committed:
drop python3-pip

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -966,6 +966,12 @@ Available commands:
+		aSOFTWARE_NAME[$software_id]='Moonlight'
+		aSOFTWARE_DESC[$software_id]='Game streaming client for Sunshine and NVIDIA GameStream'
+		aSOFTWARE_CATX[$software_id]=5
+		aSOFTWARE_DOCS[$software_id]='https://github.com/moonlight-stream/moonlight-embedded/wiki'
```
Let's use the URL to our docs even that the section needs to be created first:
```suggestion
		aSOFTWARE_DOCS[$software_id]='https://dietpi.com/docs/software/gaming/#moonlight'
```

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -11103,6 +11109,34 @@ _EOF_
+
+			# APT deps
+			aDEPS=('apt-transport-https' 'ca-certificates' 'debian-keyring' 'debian-archive-keyring' 'gnupg')
+
```
None of these is neeeded:
```suggestion
```
- APT has native HTTPS support, so `apt-transport-https` is just a transitional dummy package since Buster: https://packages.debian.org/apt-transport-https
- `ca-certificates` is necessarily pre-installed, otherwise the script would not even run until here 😉.
- `debian-keyring` is for Debian package maintainers to upload packages, never for users/admins to install packages: https://packages.debian.org/bookworm/debian-keyring
- `debian-archive-keyring` is a hard dependency of APT on Debian, while on Raspbian it is `raspbian-archive-keyring`. So the correct one is always pre-installed: https://packages.debian.org/bullseye/apt
- `gnupg` is as well pre-installed on DietPi, otherwise a bunch of other tasks and installs wouldn't work either.

mtekman replied:
Ah nice, I didn't know that about the keyrings. I thought it was how developers signed their packages.

> apt-transport-https, ca-certificates, gnupg

Yeah I figured, wasn't sure whether to be explicit or not

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -11103,6 +11109,34 @@ _EOF_
+			# APT list
+			local repo_url="$base_url/deb/raspbian"
+			G_EXEC eval "echo 'deb [signed-by=$gpg_keyring_path] $repo_url $G_DISTRO_NAME main' > $repo_path"
+			G_EXEC eval "echo 'deb-src [signed-by=$gpg_keyring_path] $repo_url $G_DISTRO_NAME main' >> $repo_path"
```
Let's align the code with other 3rd party repo installs:
```suggestion
			# Detect distro
			local distro='debian'
			(( $G_HW_MODEL < 10 )) && (( $G_RASPBIAN )) && distro='raspbian'

			# APT key
			local url='https://dl.cloudsmith.io/public/moonlight-game-streaming/moonlight-embedded/gpg.5AEE46706CF0453E.key'
			G_CHECK_URL "$url"
			G_EXEC eval "curl -sSfL '$url' | gpg --dearmor -o /etc/apt/trusted.gpg.d/dietpi-moonlight.gpg --yes"

			# APT list
			G_EXEC eval "echo 'deb https://dl.cloudsmith.io/public/moonlight-game-streaming/moonlight-embedded/$distro/ $G_DISTRO_NAME stable' > /etc/apt/sources.list.d/dietpi-moonlight.list"
```
- I'm pretty sure there is a "debian" repo path if Debian is used instead of Raspbian, which enables ARMv8 and x86_64 support.
- Using `signed-by` instead of `trusted.gpg.d` is actually the modern/better way, but I'd like to keep things aligned for now and migrate all 3rd party repos together at some point.
- Skip source package list `deb-src`, not needed for installing binary packages. If really someone wants to compile it manually, its trivial to add it manually.

mtekman replied:
Makes sense!

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -12689,6 +12723,15 @@ If no WireGuard (auto)start is included, but you require it, please do the follo
+			local gpg_keyring_path="/usr/share/keyrings/moonlight-game-streaming-moonlight-embedded-archive-keyring.gpg"
+			local repo_path="/etc/apt/sources.list.d/moonlight-game-streaming-moonlight-embedded.list"
+			[[ -f "$gpg_keyring_path" ]] && G_EXEC rm "$gpg_keyring_path"
+			[[ -f "$repo_path" ]] && G_EXEC rm "$repo_path"
```
I'm no fans of defining variables which are used only 1-2 time 😉:
```suggestion
			[[ -f '/etc/apt/sources.list.d/dietpi-moonlight.list' ]] && G_EXEC rm /etc/apt/sources.list.d/dietpi-moonlight.list
			[[ -f '/etc/apt/trusted.gpg.d/dietpi-moonlight.gpg' ]] && G_EXEC rm /etc/apt/trusted.gpg.d/dietpi-moonlight.gpg
```

mtekman replied:
agreed!

mtekman committed:
Update dietpi/dietpi-software

Co-authored-by: MichaIng <micha@dietpi.com>

mtekman committed:
Update dietpi/dietpi-software

Co-authored-by: MichaIng <micha@dietpi.com>

mtekman committed:
Update dietpi/dietpi-software

Co-authored-by: MichaIng <micha@dietpi.com>

mtekman committed:
Update dietpi/dietpi-software

Co-authored-by: MichaIng <micha@dietpi.com>

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -11103,6 +11109,27 @@ _EOF_
+			## - https://dl.cloudsmith.io/public/moonlight-game-streaming/moonlight-embedded/setup.deb.sh
+			# Detect distro
+			local distro='debian'
+			(( $G_HW_MODEL < 10 )) && (( $G_RASPBIAN )) && distro='raspbian'
```
As of: https://github.com/MichaIng/DietPi/pull/6303#issuecomment-1509859764
```suggestion
			# Enable required fake KMS driver on RPi
			(( $G_HW_MODEL > 9 )) || /boot/dietpi/func/dietpi-set_hardware rpi-opengl vc4-fkms-v3d
```

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -11113,20 +11113,17 @@ _EOF_
-			local repo_url="$base_url/deb/raspbian"
-			G_EXEC eval "echo 'deb [signed-by=$gpg_keyring_path] $repo_url $G_DISTRO_NAME main' > $repo_path"
-			G_EXEC eval "echo 'deb-src [signed-by=$gpg_keyring_path] $repo_url $G_DISTRO_NAME main' >> $repo_path"
+			G_EXEC eval "echo 'deb https://dl.cloudsmith.io/public/moonlight-game-streaming/moonlight-embedded/$distro/ $G_DISTRO_NAME stable' > /etc/apt/sources.list.d/dietpi-moonlight.list"
```
```suggestion
			G_EXEC eval "echo 'deb https://dl.cloudsmith.io/public/moonlight-game-streaming/moonlight-embedded/deb/raspbian/ $G_DISTRO_NAME main' > /etc/apt/sources.list.d/dietpi-moonlight.list"
```

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -11103,6 +11109,27 @@ _EOF_
+			## Install script adapted from
+			## - https://dl.cloudsmith.io/public/moonlight-game-streaming/moonlight-embedded/setup.deb.sh
+			# Detect distro
+			local distro='debian'
```
```suggestion
			# https://dl.cloudsmith.io/public/moonlight-game-streaming/moonlight-embedded/setup.deb.sh

```

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -966,6 +966,12 @@ Available commands:
+		aSOFTWARE_DESC[$software_id]='Game streaming client for Sunshine and NVIDIA GameStream'
+		aSOFTWARE_CATX[$software_id]=5
+		aSOFTWARE_DOCS[$software_id]='https://dietpi.com/docs/software/gaming/#moonlight'
 
```
```suggestion
		# ARM only
		(( $G_HW_ARCH > 3 )) && aSOFTWARE_AVAIL_G_HW_ARCH[$software_id,$G_HW_ARCH]=0
		# No Bookworm packages yet and Bullseye packages depend on libssl1.1
		aSOFTWARE_AVAIL_G_DISTRO[$software_id,7]=0

```

MichaIng committed:
Update dietpi/dietpi-software

MichaIng committed:
Update dietpi/dietpi-software

MichaIng committed:
Update dietpi/dietpi-software

MichaIng committed:
Update dietpi/dietpi-software

MichaIng committed:
Merge branch 'dev' into moonlight

MichaIng committed:
v8.17

- DietPi-Software | Moonlight: Add meta info

MichaIng committed:
Merge branch 'dev' into moonlight

MichaIng committed:
Merge branch 'dev' into pr/6303

MichaIng committed:
v8.17

- DietPi-Software | Moonlight: As the packages from the used repository are for RPi only, disable on all other devices for now. We may add own builds for other hardware in the future. Also install libcec, tried to be loaded when starting streaming.

MichaIng committed:
v8.17

- DietPi-Software | Moonlight: Assure that headless mode is disabled and default GPU memory applied on install

MichaIng committed:
Merge branch 'dev' into pr/6303

MichaIng committed:
v8.17

- DietPi-Software | Moonlight: The armhf binary packages are currently not compatible with ARMv6: https://github.com/moonlight-stream/moonlight-embedded/issues/832

MichaIng committed:
v8.17

- DietPi-Software | Moonlight: ARMv8 (64-bit) packages are available on Bullseye only

MichaIng committed:
Merge branch 'dev' into pr/6303

MichaIng committed:
Merge branch 'dev' into pr/6303

MichaIng committed:
v8.17

- DietPi-Software | Moonlight: Update to use ID 207, as 206 is now used by openHAB already

MichaIng committed:
v8.17

- DietPi-Software | Moonlight: Fix libcec install. It is not necessarily needed but mutes an error during Moonlight startup and of course might enable some HDMI control features
